pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ClientBuilder.java

```diff
@@ -263,7 +263,7 @@ ClientBuilder authentication(String authPluginClassName, Map<String, String> aut
      * Increasing this parameter may improve throughput when using many producers over a high latency connection.
      *
      * @param connectionsPerBroker
-     *            max number of connections per broker (needs to be greater than 0)
+     *            max number of connections per broker (needs to be not less than 0)
```

Comment:
Given the name of this setting (`connectionsPerBroker`), it isn't immediately clear to me why 0 is a valid setting. I think we should add a note specifying that a value of `0` will disable connection pooling. I think it could also be valuable to update the client configuration file with the same note: https://github.com/apache/pulsar/blob/b557e2479c70631fa0dc34606bc5492f73cfef96/pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java#L122-L126

Nit: my preference for this line is to write this as `needs to be greater than or equal to 0`. Your change is certainly correct, so I'm not sure how much my preference matters, but based on a quick search through the project, we often use `greater than or equal to` in this situation.

Reply:
Good suggestion, done. PTAL